drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -236,6 +236,31 @@ def denied_pending_referral_postings(**args)
       scoped_referral_postings(postings, **args)
     end
 
+    field :merge_candidates, Types::HmisSchema::ClientMergeCandidate.page_type, null: false
+    def merge_candidates
+      raise 'not allowed' unless current_user.can_merge_clients?
+
+      destination_id_to_source_ids = GrdaWarehouse::WarehouseClient.
+        where(data_source_id: current_user.hmis_data_source_id).
+        joins(:source). # drop non existent source clients
+        group(:destination_id).
+        having('count(*) > 1').
+        select('"destination_id", array_agg("source_id") as source_ids').
+        map { |r| [r.destination_id, r.source_ids] }.
+        to_h
```

Comment:
This looks pretty good to me, but only limiting to the data source does feel a bit dangerous.  Alternatively, you could `group(:destination_id).having('count(*) > 1').pluck(:destination_id)` and then just loop over batches of those destination ids to get the source ids.  That, at least would prevent ever selecting tens of thousands at a time if there were many duplicates.

Reply:
oh actually, I think there is a better way to do this. If the `object` being resolved is a `GrdaWarehouse::Hud::Client`, then we can use the dataloader to load `source_clients`, and use the usual scope pagination rather than loading things into arrays. I am going to try that.